server/libs/infra/src/repositories/partner.repository.ts

```diff
@@ -0,0 +1,63 @@
+import { IPartnerRepository, PartnerDirection } from '@app/domain';
+import { Injectable, Logger } from '@nestjs/common';
+import { InjectRepository } from '@nestjs/typeorm';
+import { Repository } from 'typeorm';
+import { PartnerEntity } from '../entities';
+
+@Injectable()
+export class PartnerRepository implements IPartnerRepository {
+  readonly logger = new Logger(PartnerRepository.name);
+
+  constructor(
+    @InjectRepository(PartnerEntity)
+    private readonly repository: Repository<PartnerEntity>,
+  ) {}
+
+  getAll(userId: string, direction: PartnerDirection): Promise<PartnerEntity[]> {
+    if (direction === 'shared-by') {
+      return this.repository
+        .createQueryBuilder('partners')
+        .select('partners.sharedBy', 'sharedBy')
+        .addSelect('partners.sharedWith', 'sharedWith')
+        .addSelect('partners.createdAt', 'createdAt')
+        .addSelect('partners.updatedAt', 'updatedAt')
+        .addSelect('partners.id', 'id')
+        .where('"sharedById" = :sharedBy', { sharedBy: userId })
+        .getRawMany();
+    } else {
+      return this.repository
+        .createQueryBuilder('partners')
+        .select('partners.sharedBy', 'sharedBy')
+        .addSelect('partners.sharedWith', 'sharedWith')
+        .addSelect('partners.createdAt', 'createdAt')
+        .addSelect('partners.updatedAt', 'updatedAt')
+        .addSelect('partners.id', 'id')
+        .where('"sharedWithId" = :sharedWith', { sharedWith: userId })
+        .getRawMany();
+    }
+  }
+
+  async get(sharedBy: string, sharedWith: string): Promise<PartnerEntity | null> {
+    const entity =
+      (await this.repository
+        .createQueryBuilder('partners')
+        .select('partners.sharedBy', 'sharedBy')
+        .addSelect('partners.sharedWith', 'sharedWith')
+        .addSelect('partners.createdAt', 'createdAt')
+        .addSelect('partners.updatedAt', 'updatedAt')
+        .addSelect('partners.id', 'id')
+        .where('"sharedById" = :sharedBy', { sharedBy: sharedBy })
+        .andWhere('"sharedWithId" = :sharedWith', { sharedWith: sharedWith })
+        .getRawOne<PartnerEntity>()) ?? null;
+
+    return entity;
```

Comment:
We should use the normal `this.repository.find` instead of the query builder. I think you are using this because of only having a single property on the entity for the relation/id, but there are ways around that, which still make it possible to avoid the query builder.